railties/test/generators/shared_generator_tests.rb

```diff
@@ -411,9 +425,18 @@ def run_generator_using_prerelease(args)
         assert_equal "install", @bundle_commands[0]
         assert_match "lock --add-platform", @bundle_commands[1]
 
+        # AppBase#run_bundle will add an extra platform for Apple Silicon Macs
+        # So we have to bump the bundle command index
+        # https://github.com/rails/rails/blob/2e4061666ba74cc2b2291152173e7e3a40e28450/railties/lib/rails/generators/app_base.rb#L689-L690
+        if RUBY_PLATFORM.start_with?("arm64")
```

Comment:
:+1:

Reply:
This conditional feels gross. At absolute minimum we should continue the surrounding pattern and check that the extra command is the one we're expecting... but I'd much rather just not.

Do we just want `@bundle_commands.last`? Or otherwise, can we shift out the commands as we match them, and make this one optional in that process?

Reply:
👍 Updated to use shift in a5c7fb86fb185b00d24a83cf3521619ccbd898e1

Reply:
I might be wrong, but I think Matthew's suggestion was to remove the conditional entirely by adding something like 

```ruby 
@bundle_commands.shift while @bundle_commands.first.start_with?("lock --add-platform=")
```

before the final `assert_match`.

Another alternative would be to `join` the commands:

```ruby 
assert_match %r"^exec rails (?:plugin )?new #{Regexp.escape Shellwords.join(expected_args)}", @bundle_commands.join
```

Though that would have less of a guarantee with respect to ordering.

Reply:
I thought it's better to keep the assertions, since this is the only place which covers this behavior:
https://github.com/rails/rails/blob/344e59e1d1ca92703aef3947ef1e0e28abb6709c/railties/lib/rails/generators/app_base.rb#L686-L690

Maybe it could be made into a separate test though?

Reply:
@zzak I've fixed this particular issue in #50169 and added you as a co-author.

Reply:
Thanks! I will rebase this out then when I get a chance. :bow: